### ISSUE TITLE
Detect if download failed

### ISSUE THREAD

**Issue by JohnVeness** [dojafoja/FunKii-UI#7]
If a download fails for some reason, e.g. the "Attempt 2 of 4" thing reaches its limit, at the moment the title is removed from the download queue and the next title attempted. Unless the user scrolls through the terminal output, they would not notice that the download had failed. Can you add some feedback to the GUI if this happens, maybe turning the entry in the download list red or something?

**Comment by dojafoja** [comment]
The reason it doesn't detect bad downloads in its current state is because I call FunKiiU's process_title_id() externally. The way it is written is that it simply returns when the function call is complete without specifying a return value. In python, a return without a value implicitly returns None. Therfore whether it downloaded successfully or not, it returns None, which evaluates to False. There are many issues with the fact that FunKiiU wasnt written with a GUI in mind. It is also why I don't currently show download progress in the GUI itself, because FunKiiU doesn't report that back to me. I have started modifying FunKiiU for better integration and I have these particular features working great already. But because I'm currently trying to add so many features at once, it is currently mostly broken. I haven't pushed most of my recent additions here quite yet but I have created new branches for the project, they're just broken until I push my working files. I'm working on several things and I can't give a timeline when I will be done because of work and lack of interest but I do plan on fixing these few issues and also a few different ones. Thanks for the feedback
